plugins/trace/trace_main.ml

```diff
@@ -83,20 +83,21 @@ module Cmdline = struct
       err_formatter fmt
 
   let () =
-    Config.when_ready (fun {Config.get=(!)} ->
-        match main !dump !load with
-        | Ok `Done -> ()
-        | Ok `Exit -> exit 0
-        | Error e -> match e with
-          | `Protocol_error err ->
-            exitf "Protocol error: %a" Error.pp err
-          | `System_error err ->
-            exitf "System error: %s" @@ Unix.error_message err
-          | `No_provider ->
-            exitf "No provider for the given URI"
-          | `Ambiguous_uri ->
-            exitf "More than one provider for a given URI"
-          | exception Incompatibe_args ->
-            exitf "Incompatible arguments, see usage SYNOPSIS" )
+    Future.upon Plugins.loaded (fun () ->
```

Comment:
This will not work in general... `Plugins.loaded` event will happen after all plugins are loaded, and at this time the commandline parsing may already occur, so it is too late to ask for config. Basically, we should enforce, that at the load time, each config must request the configuration (and tell us what it is looking for). 

So, the order here should be the opposite `Config.when_ready (fun {Config.get=(!)} -> Future.upon Plugins.loaded...`

Also, this example shows, that we need to expose `Config.ready : reader future` interface, for those who would like to use the future interface for building more complex evaluation strategies.